setup_cfg_fmt.py

```diff
@@ -196,6 +196,62 @@ def _python_requires(
         return _format_python_requires(minimum, excluded)
 
 
+def _install_requires(setup_cfg: str) -> List[str]:
+    cfg = configparser.ConfigParser()
+    cfg.read(setup_cfg)
+    raw = cfg.get('options', 'install_requires', fallback='')
+
+    install_requires = raw.strip().splitlines()
+    if not install_requires:
+        return []
+
+    if len(install_requires) == 1:
+        return [_normalize_req(install_requires[0])]
+
+    has_env, has_no_env = [], []
```

Comment:
@asottile instead of creating two lists I can do this with a more complicated sort function.. lmk

```diff
diff --git a/setup_cfg_fmt.py b/setup_cfg_fmt.py
index 04676f8..c60e612 100644
--- a/setup_cfg_fmt.py
+++ b/setup_cfg_fmt.py
@@ -1,9 +1,11 @@
 import argparse
 import configparser
+import functools
 import glob
 import io
 import os.path
 import re
+from typing import Callable
 from typing import Dict
 from typing import List
 from typing import Match
@@ -208,21 +210,27 @@ def _install_requires(setup_cfg: str) -> List[str]:
     if len(install_requires) == 1:
         return [_normalize_req(install_requires[0])]
 
-    has_env, has_no_env = [], []
-    for req in install_requires:
-        normalized = _normalize_req(req)
-        if ';' in normalized:
-            has_env.append(normalized)
-        else:
-            has_no_env.append(normalized)
-
     return [
         '',
-        *sorted(has_no_env),
-        *sorted(has_env),
+        *sorted(
+            (_normalize_req(req) for req in install_requires),
+            key=_sort_key,
+        ),
     ]
 
 
+def _sort_key(req: str) -> Callable[[str], int]:
+    def cmp(r1: str, r2: str) -> int:
+        if ';' in r1 and ';' not in r2:
+            return 1
+        elif ';' in r2 and ';' not in r1:
+            return -1
+        else:
+            return (r1 > r2) - (r1 < r2)
+
+    return functools.cmp_to_key(cmp)(req)
+
+
 def _normalize_req(req: str) -> str:
     name, _, envs = req.partition(';')
     normalized = _normalize_name(name)
```

Reply:
I think the two lists is fine

Reply:
a much simpler key though hehehe

```python
def key(req):
    return (';' in req, req)
```

Reply:
wow

Reply:
Well if it's that easy then I'll just do that

Reply:
lel